kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -647,14 +826,21 @@ compareForEquals first second
     | isConstructorLike second = GT
     | otherwise = compare first second
 
-bytesDifferent ::
-    MonadUnify unifier =>
+-- | Matches two constant byte values with distinct values.
+matchBytesDifferent ::
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName ->
-    MaybeT unifier (Pattern RewritingVariableName)
-bytesDifferent
-    (Recursive.project -> _ :< InternalBytesF (Const bytesFirst))
-    (Recursive.project -> _ :< InternalBytesF (Const bytesSecond))
-        | bytesFirst /= bytesSecond =
-            return Pattern.bottom
-bytesDifferent _ _ = empty
+    Maybe ()
+matchBytesDifferent first second
+    | _ :< InternalBytesF (Const bytesFirst) <- Recursive.project first
+      , _ :< InternalBytesF (Const bytesSecond) <- Recursive.project second
+      , bytesFirst /= bytesSecond =
+        Just ()
+    | otherwise = Nothing
+{-# INLINE matchBytesDifferent #-}
+
+bytesDifferent ::
+    MonadUnify unifier =>
+    unifier (Pattern RewritingVariableName)
+bytesDifferent =
+    return Pattern.bottom
```

Comment:
This code would be simplified if we did not assume that the `Bytes` are different, but checked.

Reply:
I don't understand, is this not being done on line 837?